flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/maintenance/api/DeleteOrphanFiles.java

```diff
@@ -0,0 +1,358 @@
+  static final String SKIP_ON_ERROR_TASK_NAME = "Skip On Error";
+
+  public static DeleteOrphanFiles.Builder builder() {
+    return new DeleteOrphanFiles.Builder();
+  }
+
+  private DeleteOrphanFiles() {
+    // Do not instantiate directly
+  }
+
+  public static class Builder extends MaintenanceTaskBuilder<DeleteOrphanFiles.Builder> {
+    private String location = null;
+    private Duration minAge = Duration.ofDays(3);
+    private int planningWorkerPoolSize = 10;
+    private int deleteBatchSize = 1000;
+    private boolean caseSensitive = false;
+    private int maxListingDepth = 3;
+    private int maxListingDirectSubDirs = 10;
+    private boolean usePrefixListing = false;
+    private Map<String, String> equalSchemes =
+        Maps.newHashMap(
+            ImmutableMap.of(
+                "s3n", "s3",
+                "s3a", "s3a"));
+    private final Map<String, String> equalAuthorities = Maps.newHashMap();
+    private org.apache.iceberg.actions.DeleteOrphanFiles.PrefixMismatchMode prefixMismatchMode =
+        org.apache.iceberg.actions.DeleteOrphanFiles.PrefixMismatchMode.ERROR;
+
+    /**
+     * The location to start the recursive listing the candidate files for removal. By default, the
+     * {@link Table#location()} is used.
+     *
+     * @param newLocation the task will scan
+     * @return for chained calls
+     */
+    public Builder location(String newLocation) {
+      this.location = newLocation;
+      return this;
+    }
+
+    /**
+     * Whether to use prefix listing when listing files from the file system.
+     *
+     * @param newUsePrefixListing true to enable prefix listing, false otherwise
+     * @return for chained calls
+     */
+    public Builder usePrefixListing(boolean newUsePrefixListing) {
+      this.usePrefixListing = newUsePrefixListing;
+      return this;
+    }
+
+    /**
+     * The maximum number of direct subdirectories to list in a single directory.
+     *
+     * @param newMaxListingDirectSubDirs the maximum number of direct sub-directories to list
+     * @return for chained calls
+     */
+    public Builder maxListingDirectSubDirs(int newMaxListingDirectSubDirs) {
+      this.maxListingDirectSubDirs = newMaxListingDirectSubDirs;
+      return this;
+    }
+
+    /**
+     * The maximum depth to recurse when listing files from the file system.
+     *
+     * @param newMaxListingDepth the maximum depth to recurse
+     * @return for chained calls
+     */
+    public Builder maxListingDepth(int newMaxListingDepth) {
+      this.maxListingDepth = newMaxListingDepth;
+      return this;
+    }
+
+    /**
+     * Action behavior when location prefixes (schemes/authorities) mismatch.
+     *
+     * @param newPrefixMismatchMode to action when mismatch
+     * @return for chained calls
+     */
+    public Builder prefixMismatchMode(
+        org.apache.iceberg.actions.DeleteOrphanFiles.PrefixMismatchMode newPrefixMismatchMode) {
```

Comment:
Using this will make it awkward for the users, as they always have to use fully qualified name, because of the name conflicts. No better idea yet, so just writing down for now.

Reply:
This compiles for me:

```java
import org.apache.iceberg.actions.DeleteOrphanFiles.PrefixMismatchMode;
...
    private PrefixMismatchMode prefixMismatchMode = PrefixMismatchMode.ERROR;
...
    public Builder prefixMismatchMode(PrefixMismatchMode newPrefixMismatchMode) {
      this.prefixMismatchMode = newPrefixMismatchMode;
      return this;
    }
```

Then, from another class, this works as well:

```java
import org.apache.iceberg.actions.DeleteOrphanFiles.PrefixMismatchMode;
...
      DeleteOrphanFiles.builder().prefixMismatchMode(PrefixMismatchMode.IGNORE);
```